src/stream/src/task/barrier_manager/managed_state.rs

```diff
@@ -112,6 +115,29 @@ impl ManagedBarrierState {
         }
     }
 
+    /// Remove stop barrier (epoch < `curr_epoch`), and send err.
+    pub(crate) fn remove_stop_barrier(&mut self, curr_epoch: u64) {
+        let stop_barrier = self
+            .epoch_barrier_state_map
+            .drain_filter(|k, _| k < &curr_epoch);
+        stop_barrier.for_each(|(k, v)| {
+            assert!(k < curr_epoch);
+            if let Issued {
+                collect_notifier, ..
+            } = v
+            {
+                if collect_notifier
+                    .send(Err(RwError::from(InternalError(
+                        "fail before this epoch".to_string(),
+                    ))))
+                    .is_err()
+                {
+                    warn!("fail stop barrier with curr_epoch{}", k)
+                }
+            }
+        })
```

Comment:
We don't need to send Err via `collect_notifier` because:
1. Collection of barriers in `epoch_barrier_state_map` must not been finished.
2. No one is waiting on `collect_rx` of `epoch` < `curr_epoch` since we will first finish collecting `current_epoch` before calling `remove_stop_barrier`. Byt this time, if someone is waiting on older epochs, it must have been finished.